### ISSUE TITLE
Typeof is not correct

### ISSUE THREAD

**Issue by GoogleCodeExporter** [shans/ristretto-js#3]
```
What steps will reproduce the problem?
1. var id = T("id :: String -> String", function (a) { return a; });
id(new String("a"));

What is the expected output? What do you see instead?
Throws TypeError and is not supposed to.
```

Original issue reported on code.google.com by `sa...@google.com` on 3 Feb 2012 at 1:05

**Comment by GoogleCodeExporter** [comment]
Original comment by `sa...@google.com` on 3 Feb 2012 at 3:57
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision e157a77a45a9.
```

Original comment by `sa...@google.com` on 3 Feb 2012 at 3:57
